emily/handler/src/api/handlers/chainstate.rs

```diff
@@ -104,29 +132,14 @@ pub async fn set_chainstate(
         context: EmilyContext,
         body: SetChainstateRequestBody,
     ) -> Result<impl warp::reply::Reply, Error> {
-        // Create new chainstate table entry.
-        let new_chainstate_entry: ChainstateEntry = ChainstateEntry {
-            key: ChainstateEntryKey {
-                height: body.stacks_block_height,
-                hash: body.stacks_block_hash,
-            },
-        };
-
-        // TODO(TBD): Tactfully handle state inconsistencies and race conditions across
-        // multiple requests.
-        //
-        // - chain tip is ahead.
-        // - chain tip is far behind.
-        // - two chain tips are proposed at the same time for conflicting hashes
-        // - etc.
-        //
-        // `get_chain_tip_or_set_if_absent` is inappropriate in the long run and should be
-        // deleted.
-
-        accessors::add_chainstate_entry(&context, &new_chainstate_entry).await?;
-        let response: Chainstate = new_chainstate_entry.into();
+        // Convert body to the correct type.
+        let chainstate: Chainstate = body as Chainstate;
```

Comment:
I figured this setup was the clearest to read, but I do know that nothing actually happens here. All of these are type definitions so for them to follow all the same idioms of requests and responses it's always going to be doing impactless casting.

Reply:
Ahh yeah I can see how it's easier to read. I was a little confused at first since I wouldn't think you can do a cast like this without some more fanfare (I thought you needed to use `transmute` or whatever).

I think you should do `Chainstate::from(body)` here. It's cleaner, it's also a no-op, and it paves the way for when/if we differentiate the `SetChainstateRequestBody` type from the `Chainstate` type.